### PR TITLE
Add formula from art.1.5.2.106a from NEN 9997-1+C2:2017

### LINKED ISSUE #103: [✨ Feature request]: Add formula from art.1.5.2.106a from NEN 9997-1+C2:2017
### Description

Add formula from art.1.5.2.106a from NEN 9997-1+C2:2017.

### Acceptance Criteria

Formula 1.0.1
Unit test
full docstring

### Dependencies

_No response_

### Confirmation

- [X] I have checked that this feature does not already exist in the issue tracker.
- [X] I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by PetervWestrienen**
## Description

Add formula from art.1.5.2.106a from NEN 9997-1+C2:2017.

Fixes #103 

## Problem
This formula isn't numbered, however it should still be in blueprints. My proposal is to number these formulas with three numbers: a.b.c.
a: the number of the chapter the formula is in.
b: the number of the next numbered formula minus 1.
c: the number of the formula within the section between two numbered formulas.

e.g. number of the fifth non-numbered formula between 9.3 and 9.4 will be 9.3.5
e.g. number of the second formula before the first formula of chapter 8 will be 8.0.2

In this way the formulas are different to numbered formulas which only have two numbers (formula a.b), but they are also logically ordered. This way of numbering is also consistent as the numbers of numbered formulas will never be the same as non numbered formulas.

@Blueprints-org/maintainers any thoughts on this ?

image

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by PetervWestrienen** [comment]
This formula isn't numbered, however it should still be in blueprints. My proposal is to number these formulas with three numbers: a.b.c.
a: the number of the chapter the formula is in.
b: the number of the next numbered formula minus 1.
c: the number of the formula within the section between two numbered formulas.

e.g. number of the fifth non-numbered formula between 9.3 and 9.4 will be 9.3.5
e.g. number of the second formula before the first formula of chapter 8 will be 8.0.2

In this way the formulas are different to numbered formulas which only have two numbers (formula a.b), but they are also logically ordered. This way of numbering is also consistent as the numbers of numbered formulas will never be the same as non numbered formulas.

**Comment by egarciamendez** [comment]
> This formula isn't numbered, however it should still be in blueprints. My proposal is to number these formulas with three numbers: a.b.c. a: the number of the chapter the formula is in. b: the number of the next numbered formula minus 1. c: the number of the formula within the section between two numbered formulas.
> 
> e.g. number of the fifth non-numbered formula between 9.3 and 9.4 will be 9.3.5 e.g. number of the second formula before the first formula of chapter 8 will be 8.0.2
> 
> In this way the formulas are different to numbered formulas which only have two numbers (formula a.b), but they are also logically ordered. This way of numbering is also consistent as the numbers of numbered formulas will never be the same as non numbered formulas.

I like this way of doing things. What do you guys think @Blueprints-org/maintainers?

**Review by egarciamendez — APPROVED** [review]
Great implementation! 🚀 Could you please document this approach somewhere in the docs?

File: blueprints/codes/eurocode/nen_9997_1_c2_2017/chapter_1_general_rules/formula_1_0_1.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        NEN 9997-1+C2:2017 art.1.5.2.106a - Formula (1.0.1)
+
+        Parameters
+        ----------
+        a : M
+            [:math:`a`] minor dimension of the largest cross-section at the pile tip [m].
+        b : M
+            [:math:`b`] major dimension of the largest cross-section at the pile tip [m].
+            Where: b ≤ 1.5 * a
```
Comment (bro-wi) [inline blueprints/codes/eurocode/nen_9997_1_c2_2017/chapter_1_general_rules/formula_1_0_1.py:27]: this condition is not checked in the code

File: tests/codes/eurocode/nen_9997_1_c2_2017/chapter_1_general_rules/test_formula_1_0_1.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        with pytest.raises(LessOrEqualToZeroError):
+            Form1Dot0Dot1EquivalentPilePointCenterline(a=a, b=b)
+
+    def test_raise_error_if_a_is_negative(self) -> None:
+        """Test that a NegativeValueError is raised when a negative value is passed for a."""
+        a = -0.3
+        b = 0.45
+
+        with pytest.raises(LessOrEqualToZeroError):
+            Form1Dot0Dot1EquivalentPilePointCenterline(a=a, b=b)
```
Comment (bro-wi) [inline tests/codes/eurocode/nen_9997_1_c2_2017/chapter_1_general_rules/test_formula_1_0_1.py:46]: use pytest.mark.parametrize

File: blueprints/codes/eurocode/nen_9997_1_c2_2017/chapter_1_general_rules/formula_1_0_1.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        self.a = a
+        self.b = b
+
+    @staticmethod
+    def _evaluate(
+        a: M,
+        b: M,
+    ) -> M:
+        """Evaluates the formula, for more information see the __init__ method."""
+        raise_if_negative(b=b)
```
Comment (bro-wi) [inline blueprints/codes/eurocode/nen_9997_1_c2_2017/chapter_1_general_rules/formula_1_0_1.py:39]: allowing a b that is zero doesn't make sense IMO

**Review by bro-wi — COMMENTED** [review]
Nice recommendation on the numbering of the formula. I think we should follow that.

some minor comments on the code.

**Comment by egarciamendez** [comment]
I have implemented latex to the formula and fixed some issues still present. @bro-wi can you have a look and merge?

File: blueprints/codes/eurocode/nen_9997_1_c2_2017/chapter_1_general_rules/formula_1_0_1.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        self.b = b
+
+    @staticmethod
+    def _evaluate(
+        a: M,
+        b: M,
+    ) -> M:
+        """Evaluates the formula, for more information see the __init__ method."""
+        raise_if_less_or_equal_to_zero(a=a, b=b)
+        b_calc = min(b, 1.5 * a)
```
Comment (bro-wi) [inline blueprints/codes/eurocode/nen_9997_1_c2_2017/chapter_1_general_rules/formula_1_0_1.py:43]: Shouldn't we raise an error in this case?  How I read the eurocode this implies that the formula is not applicable for a shape where b > 1.5 * a. 

Now you will receive an answer always, but I think you should be warned that this shape is not supported.

Comment (egarciamendez) [reply blueprints/codes/eurocode/nen_9997_1_c2_2017/chapter_1_general_rules/formula_1_0_1.py:43]: standard ValueError will do the trick?

**Pull request merged**